File: src/cpp_siro/src/siro_node_suscriptor.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class SiroNodeSuscriptor : public rclcpp::Node
{
public:
    SiroNodeSuscriptor() : Node("siro_node_suscriptor")
    {
        // Suscribe al tópico 'chatter' con un callback
        subscription_ = this->create_subscription<std_msgs::msg::String>(
            "chatter", 10, [this](const std_msgs::msg::String::SharedPtr msg) {
                RCLCPP_INFO(this->get_logger(), "Mensaje recibido: '%s'", msg->data.c_str());
            });

        // Configura el temporizador para imprimir cada segundo
        timer_ = this->create_wall_timer(std::chrono::seconds(1),
                                         [this]() {
                                             RCLCPP_INFO(this->get_logger(), "Esperando mensajes...");
                                         });
    }

private:
    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SiroNodeSuscriptor>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/python_siro/python_siro/siro_node_suscriptor.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class SiroNodeSuscriptor(Node):

    def __init__(self):
        super().__init__('siro_node_suscriptor')
        self.subscription_ = self.create_subscription(
            String, 'chatter', self.subscriber_callback, 10)
        self.timer_ = self.create_timer(1, self.timer_callback)

    def subscriber_callback(self, msg):
        self.get_logger().info("Mensaje recibido: '%s'" % msg.data)

    def timer_callback(self):
        self.get_logger().info("Esperando mensajes...")

def main(args=None):
    rclpy.init(args=args)
    node = SiroNodeSuscriptor()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()
